radix-engine-interface/src/api/types/re_node.rs

```diff
@@ -11,6 +11,28 @@ pub enum ScryptoRENode {
     KeyValueStore,
 }
 
+// TODO: Remove when better type system implemented
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Ord, PartialOrd)]
+#[scrypto(TypeId, Encode, Decode)]
+pub enum RENodeType {
+    Bucket,
+    Proof,
+    AuthZoneStack,
+    Worktop,
+    GlobalAccount,
+    GlobalComponent,
+    GlobalResourceManager,
+    GlobalPackage,
+    GlobalEpochManager,
```

Comment:
Feel like we should represent globals as `Global(EntityTypeId)`.

Reply:
Not sure if that gives any benefit at the moment? I'd rather avoid complexity until we get to the point of refactoring this area.

Reply:
More from representation accuracy perspective. None of the variants are really a `RENodeType`.